doc/designs/passkeys.md

```diff
@@ -0,0 +1,246 @@
+attribute, but this attribute has a special handling (a new fingerprint is
+calculated for each public key and added into the attribute sshpubkeyfp)
+which makes it unsuitable for storing values that are not keys.
+
+The attribute is multi valued, optional.
+
+```
+dn: uid=idmuser,cn=users,cn=accounts,dc=ipa,dc=test
+uid: idmuser
+...
+objectClass: top
+objectClass: person
+...
+objectClass: ipapasskeyuser
+ipapasskey: passkey:9S87qLk8/RxYJ3skwwYduomAM+/HDtz41N0+w/vRL6aGKJkLMsg+2OhO0E8pK5DuO1KmdK61K8PmH7jiYuOqbg==,9YE1s/f7J47h2A/DXCVFWulqoBXFzCSxcbGEBadkpSUFjwUudhPLnPUTv2qNamakXJgRYCZQ7vpS/t5zXMLnkw==
+```
+
+The passkey mapping has the format `passkey:credentialid,pubkey`. credential
+ID and public key are obtained during the registration phase, for instance
+by calling SSSD helper process `sssctl passkey-exec --register` or the IPA Command
+`ipa user-add-passkey LOGIN --register`.
+
+
+### Access control
+
+#### Permissions
+
+- New permission created for writing the passkey configuration:
+`System: Modify Passkey Configuration`. Granted to the Privilege `Passkey Administrators`
+- New permission created for reading the passkey configuration:
+`System: Read Passkey configuration`. Granted to all authenticated users.
+- New permission for managing passkey mapping:
+`System: Manage Passkey Mappings`. Granted to the Privilege: `Passkey Administrators`
+
+- Extend existing permission" `System: Read User Addressbook Attributes`:
+allow read access to the ipapasskey attribute (granted to all authenticated
+users). This attribute is not sensitive as it contains only public data.
+
+#### Self-service Permission
+
+- New self-service permission for managing their own passkey mapping:
+`Users can manage their own passkey mappings`
+
+#### Privilege
+
+- New privilege `Passkey Administrators` with the permissions `System: Modify Passkey Configuration` and `System: Manage Passkey Mappings`.
+
+By default only members of the admins group are allowed to modify the passkey
+settings or another user's passkeys.
+
+## Implementation
+
+### LDAP schema
+
+New objectclass and attribute for the passkey configuration object:
+```
+attributeTypes: ( 2.16.840.1.113730.3.8.24.26 NAME 'ipaRequireUserVerification' DESC 'require passkey user verification' SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE X-ORIGIN 'IPA v4.10')
+objectclasses: ( 2.16.840.1.113730.3.8.24.8 NAME 'ipaPasskeyConfigObject' DESC 'IPA passkey global config options' AUXILIARY MUST ipaRequireUserVerification X-ORIGIN 'IPA v4.10')
+```
+
+New objectclass and attribute for the passkey mapping:
+```
+attributeTypes: ( 2.16.840.1.113730.3.8.24.27 NAME 'ipapasskey' DESC 'Passkey mapping' EQUALITY caseExactMatch SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 X-ORIGIN 'IPA v4.10' )
+objectclasses: ( 2.16.840.1.113730.3.8.24.9 NAME 'ipaPasskeyUser' DESC 'IPA passkey user' AUXILIARY MAY ipapasskey X-ORIGIN 'IPA v4.10')
+```
+
+### Indices
+
+No need to add a new index for ipapasskey as the search performed by SSSD
```

Comment:
I am a bit concerned about this because 389-ds rejects unindexed searches. We don't set `require_index` during deployment so it is switched off by default but we already had issues with unindexed searches in past when the feature has made into 389-ds and was switched on by default.

Reply:
I'm not sure I understand your concern here. If the search filter is using uid and the ipapasskey attribute is in the list of returned attributes, ipapasskey index is not needed as far as I understand.

Reply:
Yes, you are right here, if ipapasskey attribute is not searched directly in the filter, we have no problem.